extensions/binding/org.eclipse.smarthome.binding.bosesoundtouch/src/main/java/org/eclipse/smarthome/binding/bosesoundtouch/handler/BoseSoundTouchHandler.java

```diff
@@ -470,4 +483,65 @@ public void refreshPresetChannel() {
                 .sorted(Comparator.comparing(StateOption::getValue)).collect(Collectors.toList());
         stateOptionProvider.setStateOptions(new ChannelUID(getThing().getUID(), CHANNEL_PRESET), stateOptions);
     }
+
+    public void handleGroupUpdated(BoseSoundTouchConfiguration masterPlayerConfiguration) {
+        String deviceId = getMacAddress();
+
+        if (masterPlayerConfiguration != null) {
+            // Stereo pair
+            if (Objects.equals(masterPlayerConfiguration.macAddress, deviceId)) {
+                groupMasterHandler = this;
+
+                if (getThing().getThingTypeUID().equals(BST_10_THING_TYPE_UID)) {
+                    // change thing type to Stereo Pair
+                    logger.debug("{}: Stereo Pair was created and this is the master device. Changing ThingType to: {}",
+                            getDeviceName(), BST_GROUP_THING_TYPE_UID);
+                    changeThingType(BST_GROUP_THING_TYPE_UID, getConfig());
+                } else if (getThing().getThingTypeUID().equals(BST_GROUP_THING_TYPE_UID)) {
+                    // updated group, this is master device, thing type is correct - nothing more to do
+                } else {
+                    // unsupported thing type - LOG
+                    logger.debug("{}: Unsupported operation for player of type: {}", getDeviceName(),
+                            getThing().getThingTypeUID());
+                }
+            } else {
+
+                thingRegistry.stream().filter(t -> {

```

Comment:
Unfortunately `thingRegistry` is already deprecated and will be removed in the light of https://github.com/eclipse/smarthome/issues/5182.
The binding should instead keep track of all its handlers itself and look up the pair player from this list. The Sonos binding should also be refactored in the near future to reflect this requirement.

Reply:
Based on the discussion of #5182 - there isn't any solution to the particular use case proposed yet, so I would suggest to leave this code as is and add the Bose binding to the list of the bindings needing an update (together with the Sonos binding).

Reply:
Despite the usage of a deprecated fields I see another potential problem here: The `groupMasterHandler` field will hold a reference to a thing handler which might in the meantime become absent. This could also be handled by the following proposal:

The `BoseSoundTouchHandlerFactory` could track all created/disposed handlers in a separate instance of a new `...HandlerObserver`. This can be shared with the `BoseSoundTouchHandler` so that the specific handler has access to all other available `BoseSoundTouchHandler`s. When pairing you could search this list and set the `groupMasterHandler`. Also, the `BoseSoundTouchHandler` should register as a listener to the new `...HandlerObserver` and get notified about removal of another handler. In case the `groupMasterHandler` reference is removed the pairing could be cleared again.

wdyt?